borg/testsuite/archiver.py

```diff
@@ -1590,6 +1590,22 @@ def test_extra_chunks(self):
         self.cmd('check', self.repository_location, exit_code=0)
         self.cmd('extract', '--dry-run', self.repository_location + '::archive1', exit_code=0)
 
+    def test_verify_data(self):
+        shutil.rmtree(self.repository_path)
+        self.cmd('init', self.repository_location)
+        self.create_src_archive('archive1')
+        archive, repository = self.open_archive('archive1')
+        with repository:
+            for item in archive.iter_items():
+                if item[b'path'].endswith('testsuite/archiver.py'):
+                    chunk = item[b'chunks'][-1]
+                    data = repository.get(chunk.id) + b'1234'
+                    repository.put(chunk.id, data)
+                    break
+            repository.commit()
+        self.cmd('check', self.repository_location, exit_code=0)
+        self.cmd('check', '--verify-data', self.repository_location, exit_code=1)
```

Comment:
Ah, right. Maybe leave now as is.

It could be later extended to make a pass over all archives and print archives / files that reference the defect chunks.